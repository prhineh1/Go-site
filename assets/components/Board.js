import React from 'react';
import Position from './Position';

export default class Game extends React.Component {
    render() {
        return (
            <div className="board">
            {this.props.boardState.map((rank, rankIndex) =>
                rank.map((content, fileIndex) => {
                    switch([this.props.turn, content]) {
                        case ['black', 'b']:
                            return <Position key={fileIndex}
                                        coord={[rankIndex, fileIndex]}
                                        content={content}
                                        onClick={this.props.selected}
                                    />
                        case ['red', 'r']:
                            return <Position key={fileIndex}
                                        coord={[rankIndex, fileIndex]}
                                        content={content}
                                        onClick={this.props.selected}
                                    />
                        default:
                            return <Position key={fileIndex}
                                        coord={[rankIndex, fileIndex]}
                                        content={content}
                                    />
                    };
                })
            )}
        </div>
        )
    }
}
