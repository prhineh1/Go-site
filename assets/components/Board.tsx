import React, { ReactElement } from 'react';
import PropTypes from 'prop-types';
import Position from './Position';
import { activePlayer, coordContent } from '../state/actions';
import { Immutable } from '../state/state';

interface BoardProps {
  turn: activePlayer;
  boardState: Immutable<coordContent[][]>;
  selected: (coord: Immutable<number[]>) => void;
  canMoveTo: Immutable<number[][]>;
  move: (toCoords: Immutable<number[]>) => void;
}

const Board: React.FC<BoardProps> = ({
  turn, boardState, selected, canMoveTo, move,
}): ReactElement => (
  <div className="board">
    {boardState.map((rank, rankIndex) => rank.map((content, fileIndex) => {
      const cases = (turn === 'black' && content === 'b')
                    || (turn === 'red' && content === 'r');
      if (cases) {
        return (
          <Position
            key={[rankIndex, fileIndex].toString()}
            coord={[rankIndex, fileIndex]}
            content={content}
            selected={selected}
          />
        );
      }
      return (
        <Position
          key={[rankIndex, fileIndex].toString()}
          coord={[rankIndex, fileIndex]}
          content={content}
          move={
            canMoveTo.some((moves) => moves[0] === rankIndex && moves[1] === fileIndex)
              ? move
              : undefined
          }
        />
      );
    }))}
  </div>
);

Board.propTypes = {
  turn: PropTypes.oneOf<activePlayer>(['black', 'red']).isRequired,
  boardState: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.oneOf<coordContent>(['b', 'r', 1]).isRequired).isRequired).isRequired,
  selected: PropTypes.func.isRequired,
  canMoveTo:
    PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.number.isRequired).isRequired).isRequired,
  move: PropTypes.func.isRequired,
};

export default Board;
