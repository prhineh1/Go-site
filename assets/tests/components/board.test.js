import React from 'react';
import Board from '../../components/Board';
import { shallow } from 'enzyme';
import { initState } from '../fixtures/fixtures';

let wrapper, boardState, turn

beforeEach(() => {
    wrapper = shallow(<Board
        boardState={initState.boardState}
        turn={'black'}
    />);
});

test("should render Board with initState", () => {
    expect(wrapper).toMatchSnapshot();
});

//TODO: write tests for 'selected' cases