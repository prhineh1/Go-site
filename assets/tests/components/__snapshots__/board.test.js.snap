// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Board with initState 1`] = `
<div
  className="board"
>
  <Position
    content={1}
    coord={
      Array [
        0,
        0,
      ]
    }
    key="0"
  />
  <Position
    content="r"
    coord={
      Array [
        0,
        1,
      ]
    }
    key="1"
  />
  <Position
    content={1}
    coord={
      Array [
        0,
        2,
      ]
    }
    key="2"
  />
  <Position
    content="r"
    coord={
      Array [
        0,
        3,
      ]
    }
    key="3"
  />
  <Position
    content={1}
    coord={
      Array [
        0,
        4,
      ]
    }
    key="4"
  />
  <Position
    content="r"
    coord={
      Array [
        0,
        5,
      ]
    }
    key="5"
  />
  <Position
    content={1}
    coord={
      Array [
        0,
        6,
      ]
    }
    key="6"
  />
  <Position
    content="r"
    coord={
      Array [
        0,
        7,
      ]
    }
    key="7"
  />
  <Position
    content="r"
    coord={
      Array [
        1,
        0,
      ]
    }
    key="0"
  />
  <Position
    content={1}
    coord={
      Array [
        1,
        1,
      ]
    }
    key="1"
  />
  <Position
    content="r"
    coord={
      Array [
        1,
        2,
      ]
    }
    key="2"
  />
  <Position
    content={1}
    coord={
      Array [
        1,
        3,
      ]
    }
    key="3"
  />
  <Position
    content="r"
    coord={
      Array [
        1,
        4,
      ]
    }
    key="4"
  />
  <Position
    content={1}
    coord={
      Array [
        1,
        5,
      ]
    }
    key="5"
  />
  <Position
    content="r"
    coord={
      Array [
        1,
        6,
      ]
    }
    key="6"
  />
  <Position
    content={1}
    coord={
      Array [
        1,
        7,
      ]
    }
    key="7"
  />
  <Position
    content={1}
    coord={
      Array [
        2,
        0,
      ]
    }
    key="0"
  />
  <Position
    content="r"
    coord={
      Array [
        2,
        1,
      ]
    }
    key="1"
  />
  <Position
    content={1}
    coord={
      Array [
        2,
        2,
      ]
    }
    key="2"
  />
  <Position
    content="r"
    coord={
      Array [
        2,
        3,
      ]
    }
    key="3"
  />
  <Position
    content={1}
    coord={
      Array [
        2,
        4,
      ]
    }
    key="4"
  />
  <Position
    content="r"
    coord={
      Array [
        2,
        5,
      ]
    }
    key="5"
  />
  <Position
    content={1}
    coord={
      Array [
        2,
        6,
      ]
    }
    key="6"
  />
  <Position
    content="r"
    coord={
      Array [
        2,
        7,
      ]
    }
    key="7"
  />
  <Position
    content={1}
    coord={
      Array [
        3,
        0,
      ]
    }
    key="0"
  />
  <Position
    content={1}
    coord={
      Array [
        3,
        1,
      ]
    }
    key="1"
  />
  <Position
    content={1}
    coord={
      Array [
        3,
        2,
      ]
    }
    key="2"
  />
  <Position
    content={1}
    coord={
      Array [
        3,
        3,
      ]
    }
    key="3"
  />
  <Position
    content={1}
    coord={
      Array [
        3,
        4,
      ]
    }
    key="4"
  />
  <Position
    content={1}
    coord={
      Array [
        3,
        5,
      ]
    }
    key="5"
  />
  <Position
    content={1}
    coord={
      Array [
        3,
        6,
      ]
    }
    key="6"
  />
  <Position
    content={1}
    coord={
      Array [
        3,
        7,
      ]
    }
    key="7"
  />
  <Position
    content={1}
    coord={
      Array [
        4,
        0,
      ]
    }
    key="0"
  />
  <Position
    content={1}
    coord={
      Array [
        4,
        1,
      ]
    }
    key="1"
  />
  <Position
    content={1}
    coord={
      Array [
        4,
        2,
      ]
    }
    key="2"
  />
  <Position
    content={1}
    coord={
      Array [
        4,
        3,
      ]
    }
    key="3"
  />
  <Position
    content={1}
    coord={
      Array [
        4,
        4,
      ]
    }
    key="4"
  />
  <Position
    content={1}
    coord={
      Array [
        4,
        5,
      ]
    }
    key="5"
  />
  <Position
    content={1}
    coord={
      Array [
        4,
        6,
      ]
    }
    key="6"
  />
  <Position
    content={1}
    coord={
      Array [
        4,
        7,
      ]
    }
    key="7"
  />
  <Position
    content="b"
    coord={
      Array [
        5,
        0,
      ]
    }
    key="0"
  />
  <Position
    content={1}
    coord={
      Array [
        5,
        1,
      ]
    }
    key="1"
  />
  <Position
    content="b"
    coord={
      Array [
        5,
        2,
      ]
    }
    key="2"
  />
  <Position
    content={1}
    coord={
      Array [
        5,
        3,
      ]
    }
    key="3"
  />
  <Position
    content="b"
    coord={
      Array [
        5,
        4,
      ]
    }
    key="4"
  />
  <Position
    content={1}
    coord={
      Array [
        5,
        5,
      ]
    }
    key="5"
  />
  <Position
    content="b"
    coord={
      Array [
        5,
        6,
      ]
    }
    key="6"
  />
  <Position
    content={1}
    coord={
      Array [
        5,
        7,
      ]
    }
    key="7"
  />
  <Position
    content={1}
    coord={
      Array [
        6,
        0,
      ]
    }
    key="0"
  />
  <Position
    content="b"
    coord={
      Array [
        6,
        1,
      ]
    }
    key="1"
  />
  <Position
    content={1}
    coord={
      Array [
        6,
        2,
      ]
    }
    key="2"
  />
  <Position
    content="b"
    coord={
      Array [
        6,
        3,
      ]
    }
    key="3"
  />
  <Position
    content={1}
    coord={
      Array [
        6,
        4,
      ]
    }
    key="4"
  />
  <Position
    content="b"
    coord={
      Array [
        6,
        5,
      ]
    }
    key="5"
  />
  <Position
    content={1}
    coord={
      Array [
        6,
        6,
      ]
    }
    key="6"
  />
  <Position
    content="b"
    coord={
      Array [
        6,
        7,
      ]
    }
    key="7"
  />
  <Position
    content="b"
    coord={
      Array [
        7,
        0,
      ]
    }
    key="0"
  />
  <Position
    content={1}
    coord={
      Array [
        7,
        1,
      ]
    }
    key="1"
  />
  <Position
    content="b"
    coord={
      Array [
        7,
        2,
      ]
    }
    key="2"
  />
  <Position
    content={1}
    coord={
      Array [
        7,
        3,
      ]
    }
    key="3"
  />
  <Position
    content="b"
    coord={
      Array [
        7,
        4,
      ]
    }
    key="4"
  />
  <Position
    content={1}
    coord={
      Array [
        7,
        5,
      ]
    }
    key="5"
  />
  <Position
    content="b"
    coord={
      Array [
        7,
        6,
      ]
    }
    key="6"
  />
  <Position
    content={1}
    coord={
      Array [
        7,
        7,
      ]
    }
    key="7"
  />
</div>
`;
